### PR TITLE
DATAUP-120 Update narrative repo testing to include how to use karma debugger

### PR DESCRIPTION
**Pull request by eamahanna**
# Description of PR purpose/changes

Instructions for how to use the karma debugger were added to docs/testing.md. I also created a table of contents for the file and fixed some of the markdown styling. I altered the Readme.md to reference the testing files, so there isn't duplicated documentation. 

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-120
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [x] Tests pass in travis and locally 
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -1,49 +1,100 @@
...
+```
+
+***JavaScript***
+
+It can be useful to immediately see your changes in the narrative. For javascript changes, you will just have to reload the page. You can print messages to the console with `console.log`
+
+To debug using the Karma Debugger complete the following steps:
+
+1. In one terminal tab enter:
+`kbase-narrative --no-browser --NotebookApp.allow_origin="*" --ip=127.0.0.1 --port=32323`
```
Comment (ialarmedalien) [inline docs/testing.md:93]: these could be broken out into a full line, i.e.

source:
` ``` `
`kbase-narrative --no-browser --NotebookApp.allow_origin="*" --ip=127.0.0.1 --port=32323`
` ``` `

result:
```
kbase-narrative --no-browser --NotebookApp.allow_origin="*" --ip=127.0.0.1 --port=32323
```

File: docs/testing.md [hunk 2]
```diff
@@ -1,49 +1,100 @@
...
+
+***JavaScript***
+
+JavaScript tests follow the common Test Spec idiom. Here, we create a new spec file for each JavaScript module. These all live under `test/unit/spec` in roughly the same subdirectory as found under `kbase-extension/static/kbase/js`. There's an example spec in `test/unit/specTemplate.js` - you can just copy this to a new module, and modify to fit your needs.
+
+### Manual Testing and Debugging
+
+***Python***
+
+For python changes, it will require shutting down the notebook, running `scripts/install_narrative.sh -u` and then starting the notebook server up again with kbase-narrative. You can print messages to the terminal using
```
Comment (ialarmedalien) [inline docs/testing.md:79]: > then starting the notebook server up again with kbase-narrative

==> 

then starting the notebook server up again with `kbase-narrative`

File: docs/testing.md [hunk 3]
```diff
@@ -1,49 +1,110 @@
...
+- Open a second tab and enter:
+
+```
+export PATH=$PATH:./node_modules/.bin/
+```
+
+- In the second tab enter:
+
+```
+karma start test/ut/karma.conf.js --browsers=Chrome --single-run=false
```
Comment (leia-sefkin) [inline docs/testing.md:107]: path should be `test/unit/karma.conf.js`

**Review by leia-sefkin — CHANGES_REQUESTED** [review]
Some small edits, but otherwise looking good! 

Wish we could clean up the directory structure a bit - kind of confusing to have testing docs in one place, tests in another, and test config in YET ANOTHER location, but I realize that's not in the scope of this PR.

File: docs/testing.md [hunk 4]
```diff
@@ -1,49 +1,110 @@
...
-* Python:  
-`src/biokbase/narrative/tests/test.cfg`  
-In the `[users]` and `[token_files]` blocks, two sets of values are needed: test_user and private_user. They don't have any special permissions, they just need to be different users.  
+***Python***
 
-* JavaScript:  
-`test/unit/testConfig.json`  
-This just needs the path to the token file (with pre-pended slash), such as `"/test/narrativetest.tok"`  
+`src/biokbase/narrative/tests/test.cfg`
+In the `[users]` and `[token_files]` blocks, two sets of values are needed: test_user and private_user. They don't have any special permissions, they just need to be different users.
```
[the remark this answers is not in the document]
Comment (eamahanna) [reply docs/testing.md:59]: @briehl Do you know what the difference is?

File: docs/testing.md [hunk 5]
```diff
@@ -1,49 +1,110 @@
...
+  - [About](#about)
+  - [How to Run Tests](#how-to-run-tests)
+  - [Add Credentials for Tests](#add-credentials-for-tests)
+  - [Testing with Travis-CI and Coveralls](#testing-with-travis-ci-and-coveralls)
+  - [Adding Your Own Tests](#adding-your-own-tests)
+  - [Manual Testing and Debugging](#manual-testing-and-debugging)
+
+## Short version
+
+1. Install the Narrative and activate its virtualenv (if appropriate)
```
[the remark this answers is not in the document]
Comment (eamahanna) [reply docs/testing.md:25]: I think this means your are suppose to activate your conda environment i.e. `conda activate my_narrative_env`. @briehl is this accurate?

[the remark this answers is not in the document]
Comment (briehl) [reply docs/testing.md]: Yeah. This doc dates back to the days where it was easiest to use `virtualenv` to install and run things. You could probably rephrase that as just `Install the Narrative using the directions in the README` or something similar.

File: docs/testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (briehl) [reply docs/testing.md:59]: They're meant to be two different, valid, tokens from different accounts. They get that name due to how some tests get run, preventing a "test_user" from reading / writing to a Narrative owned by a "private_user".

File: docs/testing.md [hunk 6]
```diff
@@ -1,49 +1,119 @@
...
+
+Install grunt-cli via the following command:
+
+```
+npm install -g grunt-cli
+```
+
+## Short version
+
+1. Install the Narrative and activate its conda environment (if appropriate)
```
Comment (briehl) [inline docs/testing.md:25]: maybe just `activate its environment`? It doesn't HAVE to be conda, though that's what the default instructions give. I'm sure it'll work fine with venv, or poetry, or pipenv, or whatever.

**Review by briehl — APPROVED** [review]
Looks good to me! I agree with @leia-sefkin 's changes, and made a few comments inline.

File: docs/testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (eamahanna) [reply docs/testing.md:59]: @briehl Is how to get these tokens documented anywhere else in the narrative docs or general KBase docs?

[the remark this answers is not in the document]
Comment (eamahanna) [reply docs/testing.md:59]: Added link to developer token documentation

**Review by leia-sefkin — APPROVED** [review]

**Pull request merged**